ledger/participant-integration-api/src/main/scala/platform/DispatcherState.scala

```diff
@@ -24,53 +24,54 @@ class DispatcherState(dispatcherShutdownTimeout: Duration)(implicit
 ) {
   private val logger = ContextualizedLogger.get(getClass)
   @volatile private var dispatcherStateRef: DispatcherState.State =
-    DispatcherNotInitialized
+    DispatcherNotRunning
 
-  def initialized: Boolean = dispatcherStateRef match {
-    case _: DispatcherState.DispatcherOperational => true
-    case DispatcherState.DispatcherNotInitialized | DispatcherState.DispatcherShutdown => false
+  def getDispatcher: Dispatcher[Offset] = dispatcherStateRef match {
```

Comment:
why is this not synchronized too? (then we could remove the volatile)

Reply:
synchronized is more expensive than volatile access (and this method is going to be used more frequently than others) and we don't need synchronization, but just the most recent value, no?

Reply:
Isn't this used once per stream, plus once per updating it from the InMemoryStateUpdater? This sounds not too frequent to me or do I miss something?

Reply:
I also mentioned that we don't need synchronization as it might imply more complex semantics (write-after-read) like in the other methods. Anyway, this is a small detail so switching it to synchronized